codegen/src/generator/ec2.rs

```diff
@@ -0,0 +1,442 @@
+use inflector::Inflector;
+
+use botocore::{Member, Operation, Service, Shape};
+use std::borrow::Cow;
+use super::GenerateProtocol;
+use super::generate_field_name;
+
+pub struct Ec2Generator;
+
+impl GenerateProtocol for Ec2Generator {
+    fn generate_methods(&self, service: &Service) -> String {
+        service.operations.values().map(|operation| {
+            format!(
+                "{documentation}
+{method_signature} {{
+    let mut request = SignedRequest::new(
+        \"{http_method}\",
+        \"{endpoint_prefix}\",
+        self.region,
+        \"{request_uri}\",
+    );
+    let mut params = Params::new();
+
+    params.put(\"Action\", \"{operation_name}\");
+    params.put(\"Version\", \"2015-10-01\");
```

Comment:
Fixed in 18fa1f2! I used `service.metadata.api_version`, from [another `Service` struct](https://github.com/rusoto/rusoto/blob/74df8e602c90e91350b8c016b0d2f0a9cdaa7940/codegen/src/botocore.in.rs#L7).